### PR TITLE
Clock

### PR DESCRIPTION
**Pull request by Jia-A**
In this PR of Dot on Time, I have added pomodoro feature according to the focus and break minutes set by the user.

### CODE REVIEW

File: src/pages/homepage.jsx [hunk 1]
```diff
@@ -2,21 +2,22 @@ import { useState } from "react";
...
 import "../styles/homepage.css";
-// import { Link } from "react-router-dom";
+import { Link } from "react-router-dom";
 import { useTask } from "../context/task-context";
 
 const Homepage = () =>{
     const [ addModal, setAddModal ] = useState(false);
     const [currentTask, setCurrentTask] =  useState({})
     const { tasks, createTask, singleTask, setSingleTask, editing, setEditing, editTask, deleteTask } = useTask();
+    console.log(tasks)
```
Comment (kavyakunder) [inline src/pages/homepage.jsx:12]: remove console log from here.

**Review by kavyakunder — APPROVED** [review]
Code is easy to understand.
Looks good to merge!

File: src/pages/homepage.jsx (hunk 1 above)
Comment (Jia-A) [reply src/pages/homepage.jsx:12]: Yeah. will do that.

File: src/App.js [hunk 2]
```diff
@@ -14,7 +14,8 @@ function App() {
       <Route path = '/homepage' element = { <Homepage/>} />
       <Route path = "/login" element = { <Login/> } />
       <Route path = '/signup' element = { <Signup/>} />
-      <Route path = '/clock' element = { <Clock/>} />
+      {/* <Route path = '/clock' element = { <Clock/>} /> */}
```
Comment (Megha-Joshi) [inline src/App.js:17]: Remove this commented code, if not required.

File: src/pages/clock.jsx [hunk 3]
```diff
@@ -1,49 +1,118 @@
...
+    const nextTimer = () => {
+        const nextMode = modeRef.current === "work" ? "break" : "work";
+        const currentSeconds = nextMode === "work" ? currentTask.duration * 60 : currentTask.break * 60;
+        setMode(nextMode);
+        modeRef.current = nextMode;
+        setSecondsLeft(currentSeconds)
+        secondsLeftRef.current = currentSeconds;
+    }
+ 
+
```
Comment (Megha-Joshi) [inline src/pages/clock.jsx:40]: Give proper spacing.

File: src/pages/homepage.jsx (hunk 1 above)
Comment (Megha-Joshi) [reply src/pages/homepage.jsx:12]: +1

File: src/pages/homepage.jsx [hunk 4]
```diff
@@ -2,21 +2,22 @@ import { useState } from "react";
...
     }
 
     const editTaskHandler = (task) =>{
         setEditing(true);
         setAddModal(true);
-        setSingleTask({...singleTask, name : task.name, duration : task.duration, description : task.description})
+        setSingleTask({...singleTask, name : task.name, duration : task.duration, description : task.description, break : task.break})
         setCurrentTask(task)  
         
     }
```
Comment (Megha-Joshi) [inline src/pages/homepage.jsx:23]: Remove the extra spaces.

**Review by Megha-Joshi — APPROVED** [review]
- Remove consoles and commented code.
- Folder structuring is good.
- Good to merge.

**Pull request merged**